db/migrations/20220610144605_hmscontent_47.up.sql

```diff
@@ -0,0 +1,44 @@
+BEGIN;
+
+--
+-- repositories
+--
+CREATE TABLE IF NOT EXISTS repositories (
+    uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE,
+    updated_at TIMESTAMP WITH TIME ZONE,
+    url VARCHAR(255) NOT NULL,
+    last_read_time TIMESTAMP WITH TIME ZONE DEFAULT NULL,
+    last_read_error VARCHAR(255) DEFAULT NULL,
+    refer_repo_config UUID DEFAULT NULL
+);
+
+ALTER TABLE ONLY repositories
+ADD CONSTRAINT fk_repositories
+FOREIGN KEY (refer_repo_config)
+REFERENCES repository_configurations(uuid)
+ON DELETE SET NULL;
+
+--
+-- repository_rpms
+--
+CREATE TABLE IF NOT EXISTS repository_rpms (
+    uuid UUID UNIQUE NOT NULL PRIMARY KEY,
+    created_at TIMESTAMP WITH TIME ZONE,
+    updated_at TIMESTAMP WITH TIME ZONE,
+    name VARCHAR(255) NOT NULL,
+    arch VARCHAR(255) NOT NULL,
+    version VARCHAR(255) NOT NULL,
+    release VARCHAR(255),
+    epoch INTEGER DEFAULT 0 NOT NULL,
+    summary VARCHAR(255) NOT NULL,
+    description TEXT NOT NULL,
+    refer_repo UUID NOT NULL
+);
```

Comment:
I played a bit with this and started building ontop of it over here:  https://github.com/content-services/content-sources-backend/pull/38

And i realized we probably should split out rpms into a many to many association. so i envision something like:

repositories:   uuid, url
rpms:   uuid, name, version, release, arch, epoch
repository_rpms:   repository_uuid, rpm_uuid

This would allow some normalization and deduplication of rpm data

Reply:
Thanks! Updated! now it has a many-to-many relationship for repositories and rpms.